trollflow2/plugins/__init__.py

```diff
@@ -286,6 +291,28 @@ def save_datasets(job):
     ``staging_zone`` directory, such that the filename written to the
     headers remains meaningful.
 
+    The product list may contain a ``call_on_done`` parameter.
+    This parameter has effect if and only if ``eager_writing`` is False
+    (which is the default).  It should contain a list of references
+    to callables.  Upon computation time, each callable will be
+    called with four arguments: the result of ``save_dataset``,
+    targets (if applicable), the full job dictionary, and the
+    dictionary describing the format config and output filename
+    that was written.  The parameter ``targets`` is set to None
+    if using a writer where :meth:`~satpy.Scene.save_datasets`
+    does not return this.  The callables must return again the
+    ``save_dataset`` return value (possibly altered).  This callback
+    could be used, for example, to ship products as soon as they are
+    successfully produced.  Three callback functions are provided
+    with trollflow2: :func:`callback_log`, :func:`callback_move`, and
+    :func:`callback_close`.  If using the geotiff or ninjogeotiff writers,
+    :func:`callback_close` should be used before the others for correct
+    results.  When using :func:`callback_move`, the user must also set
+    ``early_moving`` to True and use a ``staging_zone``.  If both are
+    used, :func:`callback_log` must be called AFTER :func:`callback_move`,
+    because :func:`callback_log` searches for the final destination of
+    the file and reports the size (so it accesses the metadata).
```

Comment:
This could be a separate chapter.

Should the functionality of these built-in callback functions explained here? Now the docstring just says how they should be used, not what they do.

Reply:
Looking at the callback docstrings I think it's even better if there's just a (bullet) list here that links to the corresponding docstrings. The individual docstrings are easier at least for me to understand than this longer version. So something like
```
Three built-in are provided with Trollflow2: :func:`callback_close`, :func:`callback_move` and :func:`callback_log`.
```